conjur_api/models/general/credentials_data.py

```diff
@@ -20,13 +20,16 @@ class CredentialsData:
 
     # pylint: disable=too-many-arguments
     def __init__(self, machine: str = None, username: str = None, password: str = None, api_key: str = None,
-                 api_token: str = None, api_token_expiration: str = None):
+                 api_token: str = None, api_token_expiration: str = None, code: str = None, code_verifier: str = None, nonce: str = None):
```

Comment:
This is already a long list of parameters and I fear it's getting out of hand. Perhaps we should create a separate class for holding the `code`, `code_verifier`, and `nonce` values since they're logically grouped together.

Reply:
Class Client object takes ConjurConnectionInfo, ssl_verification_mode, and AuthenticationStrategyInterface as input. Authn, OIDC, and LDAP authentication mechanisms are supported by AuthenticationStrategyInterface. In addition, OIDC and other authentication methods require CredentialProvider as an input. Add new class will require lot of changes

Reply:
I don't think this would require much new code. Here's what I have in mind: https://github.com/cyberark/conjur-api-python/commit/0562f05003833757ab6b00bc36de2dc6905990d8